sdk/src/azure/platform/CMakeLists.txt

```diff
@@ -130,6 +130,7 @@ if (AZ_MQTT_TRANSPORT_IMPL STREQUAL "NONE" AND (UNIX AND NOT APPLE))
     az_mqtt_notransport 
       STATIC
       ${CMAKE_CURRENT_LIST_DIR}/az_nomqtt.c
+      ${CMAKE_CURRENT_LIST_DIR}/az_mqtt_policy.c
```

Comment:
is this needed when there's no mqtt implementation?

Reply:
Yeah... it's kind of odd but az_nomqtt.c _is_ an interface implementation. If there really isn't any MQTT we won't be compiling the stateful clients at all (since they require the interface).